Sources/InfomaniakCoreDB/TransactionExecutor.swift

```diff
@@ -0,0 +1,79 @@
+/*
+ Infomaniak Core - iOS
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import InfomaniakCore
+import RealmSwift
+
+/// Shared protected DB transaction implementation.
+///
+///  Only write transactions are protected from sudden termination, will extend to read if required.
+public struct TransactionExecutor: Transactionable {
```

Comment:
Implementation of Transactionable lives in `core-DB` with only a `RealmAccessible` dependency (getRealm())